### PR TITLE
Run `clang-format` for Examples and Tutorials

### PR DESCRIPTION
**Pull request by seladb**

### CODE REVIEW

File: Examples/ArpSpoofing/main.cpp [hunk 1]
```diff
@@ -14,59 +14,54 @@
...
+		exit(1);                                                                                                       \
+	} while (0)
+
+static struct option L3FwdOptions[] = {
+	{ "interface", required_argument, nullptr, 'i' },
+    { "victim",    required_argument, nullptr, 'c' },
+	{ "gateway",   required_argument, nullptr, 'g' },
+    { "help",      no_argument,       nullptr, 'h' },
+	{ "version",   no_argument,       nullptr, 'v' },
+    { nullptr,     0,                 nullptr, 0   }
```
Comment (seladb) [inline Examples/ArpSpoofing/main.cpp:31]: @tigercosmos for some reason clang-format mixes tabs and spaces here, I have no idea why 🤷‍♂️ 
I tried different things but couldn't make it to work. Maybe you know?

It happens in other files also, like here: https://github.com/seladb/PcapPlusPlus/blob/6990c13f0b24192b1b9230cc23227406a08a3927/Examples/IcmpFileTransfer/Common.cpp#L21

But in others it doesn't happen: https://github.com/seladb/PcapPlusPlus/blob/6990c13f0b24192b1b9230cc23227406a08a3927/Examples/DNSResolver/main.cpp#L19

Comment (seladb) [reply Examples/ArpSpoofing/main.cpp:31]: Apparently in some of the files, changing the order of the arguments fixes the issue, this is very weird 😕

File: .clang-format [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 BasedOnStyle: Microsoft
-UseTab: AlignWithSpaces
+UseTab: ForIndentation
```
Comment (seladb) [inline .clang-format:3]: @tigercosmos I think we should change this, it looks nicer, what do you think?
If you agree I can open a PR for Common++ as well

Comment (tigercosmos) [reply .clang-format:3]: I don't have a strong opinion about it. it's fine to me.

File: Examples/ArpSpoofing/main.cpp (hunk 1 above)
Comment (tigercosmos) [reply Examples/ArpSpoofing/main.cpp]: @seladb  whenever you think clang-format is stupid, you can turn off the format for that part.

**Comment by tigercosmos** [comment]
@seladb  precommit failed.  did you check if your clang-format version correct?

**Comment by seladb** [comment]
> @seladb precommit failed. did you check if your clang-format version correct?

Hmm I'm using version 18.1.6 on Windows
```cmd
D:\seladb\PcapPlusPlus\Examples>D:\Apps\LLVM\bin\clang.exe --version
clang version 18.1.6
Target: x86_64-pc-windows-msvc
Thread model: posix
```

File: .clang-format (hunk 2 above)
Comment (seladb) [reply .clang-format:3]: ok, I'll open a separate PR for this change

File: Examples/ArpSpoofing/main.cpp (hunk 1 above)
Comment (seladb) [reply Examples/ArpSpoofing/main.cpp]: It's concerning that we can't rely on `clang-format` and need to go over the code ourselves. This kind of beats the purpose of a code formatter 😕

Comment (tigercosmos) [reply Examples/ArpSpoofing/main.cpp]: @seladb this kind of stuff rarely happens. We are formatting a bunch of code at once so there might be a few weird formatting. But overall, we will get the fruits after all formatting. :)

**Comment by tigercosmos** [comment]
@seladb  remember to update `.pre-commit.config` file for the tested files.

**Comment by seladb** [comment]
> @seladb  remember to update `.pre-commit.config` file for the tested files.

Yes, that's a good point, thanks for the reminder!

File: Tests/Pcap++Test/Tests/LiveDeviceTests.cpp [hunk 3]
```diff
@@ -302,7 +302,8 @@ PTF_TEST_CASE(TestPcapLiveDevice)
 	{
 		// Should probably be refactored as PTF_ASSERT_CONTAINS or similar.
 		auto const ipAddresses = liveDev->getIPAddresses();
-		PTF_ASSERT_TRUE(std::any_of(ipAddresses.begin(), ipAddresses.end(), [ipToSearch](pcpp::IPAddress const& addr) { return addr == ipToSearch; }));
+		PTF_ASSERT_TRUE(std::any_of(ipAddresses.begin(), ipAddresses.end(),
+		                            [ipToSearch](pcpp::IPAddress const& addr) { return addr == ipToSearch; }));
```
Comment (seladb) [inline Tests/Pcap++Test/Tests/LiveDeviceTests.cpp:306]: `clang-format` failed after I included `Tests` in pre-commit

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -24,7 +24,7 @@ repos:
     hooks:
       - id: clang-format
         args: ["--style=file"]  # Use the .clang-format file for configuration
-        files: ^Common\+\+/.*\.(cpp|h)$
+        files: ^(Common\+\+|Tests|Examples)/.*\.(cpp|h)$
```
Comment (seladb) [inline .pre-commit-config.yaml:27]: Updated to cover `Common++`, `Tests` and `Examples`

**Comment by tigercosmos** [comment]
@seladb If you have gone through all the files, I will approve the PR.

**Comment by seladb** [comment]
> @seladb If you have gone through all the files, I will approve the PR.

@tigercosmos I went through all the files and made updates where necessary. However feel free to do another pass and make sure I didn't miss anything

**Review by tigercosmos — APPROVED** [review]
I did a quick overview. LGTM.

**Pull request merged**